app/services/cancel_interview.rb

```diff
@@ -0,0 +1,31 @@
+class CancelInterview
+  attr_reader :auth, :application_choice, :interview, :cancellation_reason
+
+  def initialize(
+    actor:,
+    application_choice:,
+    interview:,
+    cancellation_reason:
+  )
+    @auth = ProviderAuthorisation.new(actor: actor)
+    @application_choice = application_choice
+    @interview = interview
+    @cancellation_reason = cancellation_reason
+  end
+
+  def save!
+    auth.assert_can_make_decisions!(
+      application_choice: application_choice,
+      course_option_id: application_choice.offered_option.id,
+    )
+
+    if ApplicationStateChange.new(application_choice).can_cancel_interview?
```

Comment:
Currently, this will fail silently if you have all the permissions, but the application is not in the right state. Can we do something to let the user know that the cancellation failed?

(I think the same is true for the CreateInterview service)

Reply:
Added else condition that throws exception. Under other circumstances I would have changed the order of the actions in a transaction and let `cancel_interview!` to throw the exception but in this instance as we can only transition the state if there are no other interviews it's not possible.

Also updated `CreateInterview`.